File: micropython/Sensorkit/BicolorLed.py
from machine import Pin
# A bicolor red and green led (with 3 colors and 4 states)
class BicolorLed():
    def __init__(self, RED, GREEN):
        global pinRed, pinGreen
        pinRed = Pin(RED, Pin.OUT)
        pinGreen = Pin(GREEN, Pin.OUT)
    def off(self):
        pinRed.off()
        pinGreen.off()
    def red(self):
        pinRed.on()
        pinGreen.off()
    def green(self):
        pinRed.off()
        pinGreen.on()
    def orange(self):
        pinRed.on()
        pinGreen.on()

File: arduino/Sensorkit/src/BicolorLed.cpp
#include "BicolorLed.h"

BicolorLed::BicolorLed(int RED, int GREEN) {
  pinRed = RED;
  pinGreen = GREEN;      
}
void BicolorLed::setup() {
  pinMode(pinRed, OUTPUT);
  pinMode(pinGreen, OUTPUT);
}
void BicolorLed::off() {
  digitalWrite(pinRed, LOW);
  digitalWrite(pinGreen, LOW);
}
void BicolorLed::red() {
  digitalWrite(pinRed, HIGH);
  digitalWrite(pinGreen, LOW);
}
void BicolorLed::green() {
  digitalWrite(pinRed, LOW);
  digitalWrite(pinGreen, HIGH);
}
void BicolorLed::orange() {
  digitalWrite(pinRed, HIGH);
  digitalWrite(pinGreen, HIGH);
}
